# Hamoye ID `16010363ff01f000`

In [1]:
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt

### About Dataset
**This dataset is from the `Food and Agriculture Organization` of the United Nations.**

In [2]:
#Load dataset
food_df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv', encoding='latin-1')
food_df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [3]:
#Check the info of the dataset
food_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [4]:
#Check the shape of the dataset
food_df.shape

(60943, 12)

In [5]:
food_df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [6]:
#Check if there's any missing value
food_df.isnull().any()

Area Code       False
Area            False
Item Code       False
Item            False
Element Code    False
Element         False
Unit            False
Y2014            True
Y2015            True
Y2016            True
Y2017            True
Y2018            True
dtype: bool

In [7]:
#Sum the total of the missing value in different columns
food_df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [8]:
#Check if there's any duplicate
food_df.duplicated().sum()

0

### 1. Perform a groupby operation on ‘Element’.  What year has the highest sum of Stock Variation?

In [9]:
#perform groupby on Element with sum function
food_df.groupby('Element')[['Y2014','Y2015','Y2016','Y2017', 'Y2018' ]].sum()

Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein supply quantity (g/capita/day)    11842.45    11833.56  
Residuals                                 35500.00    34864.00  
Seed                                      24870.14    25263.14  
Stock Variation                           54316.91    20577.91  
Total Population - Both sexes           1112641.00  1140605.00  
Tourist consumption                          91.00       90.00

Answser: **Y2014 with 58749.83**

### 2. What is the total Protein supply quantity in Madagascar in 2015?

In [10]:
#Fetch the data for Madagascar
Madagascar = food_df[food_df['Area'] == 'Madagascar']

In [11]:
#view first 5 rows of Madagascar data
Madagascar.head()

Area Code        Area  Item Code         Item  Element Code  \
30715        129  Madagascar       2501   Population           511   
30716        129  Madagascar       2501   Population          5301   
30717        129  Madagascar       2901  Grand Total           664   
30718        129  Madagascar       2901  Grand Total           674   
30719        129  Madagascar       2901  Grand Total           684   

                                      Element             Unit     Y2014  \
30715           Total Population - Both sexes     1000 persons  23590.00   
30716                Domestic supply quantity      1000 tonnes      0.00   
30717           Food supply (kcal/capita/day)  kcal/capita/day   1986.00   
30718  Protein supply quantity (g/capita/day)     g/capita/day     44.45   
30719      Fat supply quantity (g/capita/day)     g/capita/day     22.18   

          Y2015     Y2016     Y2017     Y2018  
30715  24234.00  24894.00  25571.00  26262.00  
30716      0.00      0.00      0.00      0.00  
30717   1922.00   1918.00   1943.00   1938.00  
30718     43.26     43.35     43.94     44.17  
30719     22.97     23.83     25.26     24.12

In [12]:
#perform groupby on Element of Madagascar data with sum function
Madagascar.groupby('Element')['Y2015'].sum()

Element
Domestic supply quantity                  31214.98
Export Quantity                             494.75
Fat supply quantity (g/capita/day)           91.85
Feed                                       2070.22
Food                                      21120.65
Food supply (kcal/capita/day)              7685.00
Food supply quantity (kg/capita/yr)         871.59
Import Quantity                            1721.80
Losses                                     2176.00
Other uses (non-food)                      3957.12
Processing                                 1803.00
Production                                29482.89
Protein supply quantity (g/capita/day)      173.05
Residuals                                  -616.00
Seed                                        699.00
Stock Variation                            -505.00
Total Population - Both sexes             24234.00
Name: Y2015, dtype: float64

Answer: **The total Protein supply quantity in Madagascar in 2015 is 173.05**

### 3. What is the total sum of Wine produced in 2015 and 2018 respectively?

In [13]:
#To view all the rows to max and hieight of 500

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 500)

In [14]:
#perform groupby on Item with sum function
food_df.groupby(['Item'])['Y2015', 'Y2018'].sum()

C:\Users\Deedoth\AppData\Local\Temp\ipykernel_2776\3311152364.py:2: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  food_df.groupby(['Item'])['Y2015', 'Y2018'].sum()


Y2015       Y2018
Item                                                   
Alcohol, Non-Food                   2180.00     2293.00
Alcoholic Beverages                98783.72    97847.27
Animal Products                    11811.73    11578.61
Animal fats                       200675.72   269648.27
Apples and products                10559.15     9640.51
Aquatic Animals, Others                9.26        5.92
Aquatic Plants                       631.23      452.02
Aquatic Products, Other              638.42      456.65
Bananas                            55046.76    55549.23
Barley and products                29153.44    31253.70
Beans                              20315.77    20121.14
Beer                               42053.67    41612.12
Beverages, Alcoholic                2818.38     2979.21
Beverages, Fermented               47447.92    46874.94
Bovine Meat                        22766.42    23926.30
Butter, Ghee                        1165.69     1175.59
Cassava and products              410235.89   435862.14
Cephalopods                          496.95      532.07
Cereals - Excluding Beer          845669.36   903280.79
Cereals, Other                     18912.55    22125.96
Citrus, Other                      15188.91    15804.14
Cloves                                77.29       82.36
Cocoa Beans and products            7671.59     8353.83
Coconut Oil                          556.18      537.70
Coconuts - Incl Copra               6646.28     6526.12
Coffee and products                 3849.99     4009.65
Cottonseed                          7998.00     8802.00
Cottonseed Oil                      1507.90     1421.56
Cream                                233.37      271.61
Crustaceans                          732.86      833.61
Dates                              11108.74    11454.52
Demersal Fish                       6123.88     6616.23
Eggs                               21431.48    20638.48
Fats, Animals, Raw                  1729.80     1707.11
Fish, Body Oil                       193.73      219.65
Fish, Liver Oil                        6.36       48.02
Fish, Seafood                      43293.41    44810.20
Freshwater Fish                    13195.86    14244.98
Fruits - Excluding Wine           282857.19   290381.46
Fruits, Other                      57504.33    61370.30
Grand Total                       118937.92   119388.06
Grapefruit and products             2352.21     2449.18
Grapes and products (excl wine)    14607.88    13992.75
Groundnut Oil                       5561.17     5463.60
Groundnuts (Shelled Eq)            42199.35    46513.58
Honey                                713.12      681.56
Infant food                          340.38      298.01
Lemons, Limes and products          3667.51     4220.50
Maize Germ Oil                       950.64      865.77
Maize and products                287659.14   297383.10
Marine Fish, Other                  3153.59     3134.78
Meat                               66042.60    70553.53
Meat, Other                         5287.66     5412.90
Milk - Excluding Butter           266446.14   281018.64
Millet and products                40459.94    49783.34
Miscellaneous                       6803.24     6715.51
Molluscs, Other                      134.91      157.20
Mutton & Goat Meat                  9625.44     9769.46
Nuts and products                   6939.31     6886.56
Oats                                 930.02     1029.29
Offals                              7841.07     8751.58
Offals, Edible                      7841.07     8751.58
Oilcrops                          163097.03   180845.03
Oilcrops Oil, Other                 5398.90     5580.14
Oilcrops, Other                     6584.80     6464.42
Olive Oil                           1711.65     1983.76
Olives (including preserved)       12547.69    12824.75
Onions                             36973.65    40608.41
Oranges, Mandarines                33421.50    34471.68
Palm Oil                           27229.24    32732.58
Palm k

Answer: **Y2015: 4251.81, Y2018: 4039.32**

### 4. What is the mean and standard deviation across the whole dataset for the year 2017 to 2 decimal places?

In [15]:
#get the mathematical operation of year 2017
food_df.Y2017.describe()

count     59437.000000
mean        140.917765
std        1671.862359
min       -1582.000000
25%           0.000000
50%           0.100000
75%           9.000000
max      190873.000000
Name: Y2017, dtype: float64

Answer: **Mean = 140.92, Std = 1671.86

### 5. What is the total number of unique countries in the dataset?

In [16]:
#Get the number of unique value of the countries
food_df.Area.nunique()

49

### 6. Perform a groupby operation on ‘Element’.  What is the total number of the sum of Processing in 2017?

In [17]:
#perform groupby on Element with sum function
food_df.groupby('Element')[['Element', 'Y2017']].sum()

Y2017
Element                                           
Domestic supply quantity                2088198.10
Export Quantity                          182338.80
Fat supply quantity (g/capita/day)        10253.84
Feed                                     223705.68
Food                                    1258888.28
Food supply (kcal/capita/day)            454681.00
Food supply quantity (kg/capita/yr)       48690.04
Import Quantity                          294559.09
Losses                                   160614.00
Other uses (non-food)                     91645.97
Processing                               292836.00
Production                              2030056.89
Protein supply quantity (g/capita/day)    11842.45
Residuals                                 35500.00
Seed                                      24870.14
Stock Variation                           54316.91
Total Population - Both sexes           1112641.00
Tourist consumption                          91.00

Answer: **292836.0**

### 7. Which year had the least correlation with ‘Element Code’?

In [18]:
#get the correlation values
food_df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

Answer: **2016**

### 8. What is the total number and percentage of missing data in 2014 to 3 decimal places?

In [19]:
#get the number of missing value on year 2014
food_df.Y2014.isnull().sum()

1589

In [20]:
#get the percentage of the missing value as against the total number of the data
1589/food_df.shape[0]

0.026073544131401474

Answer: **Total Number = 1589, Percentage = 2.607%

### 9. Select columns ‘Y2017’ and ‘Area’, Perform a groupby operation on ‘Area’.  Which of these Areas had the highest sum in 2017?

In [21]:
#perform groupby on Area, sum it and sort the value
food_df.groupby('Area')[['Y2017', 'Area']].sum().sort_values(by='Y2017', ascending=False)

Y2017
Area                                   
Nigeria                      1483268.23
Egypt                         866379.92
South Africa                  517590.54
Ethiopia                      448683.76
Morocco                       388495.36
Ghana                         337599.06
Algeria                       325644.27
United Republic of Tanzania   322616.85
Kenya                         264660.66
Sudan                         239931.92
Cameroon                      232030.43
Angola                        229159.57
Côte d'Ivoire                 224599.01
Uganda                        213950.38
Malawi                        181098.71
Mozambique                    161407.98
Mauritania                    156665.46
Mali                          149928.33
Madagascar                    131197.73
Niger                         126707.58
Benin                         124771.22
Tunisia                       124167.20
Zambia                        103223.77
Burkina Faso                  101855.07
Guinea                         98138.87
Senegal                        95681.15
Zimbabwe                       75919.34
Rwanda                         73663.69
Chad                           71594.68
Sierra Leone                   55311.33
Eswatini                       54343.33
Mauritius                      51114.83
Togo                           49841.88
Congo                          41181.68
Central African Republic       29937.00
Namibia                        29874.89
Liberia                        29342.20
Gabon                          27979.64
Gambia                         23154.18
Djibouti                       22729.91
Botswana                       22101.30
Lesotho                        21267.96
Guinea-Bissau                  19102.77
Cabo Verde                     14650.74
Sao Tome and Principe          12662.63
Seychelles                       442.34
Comoros                           59.84
Sudan (former)                     0.00
Ethiopia PDR                       0.00

Answer: **Nigeria**

### 10. Select columns ‘Y2017’ and ‘Area’, Perform a groupby operation on ‘Area’.  Which of these Areas had the 7th lowest sum in 2017?

In [22]:
#perform groupby on Area, sum it and sort the value
food_df.groupby('Area')[['Y2017']].sum().sort_values(by='Y2017', ascending=True)

Y2017
Area                                   
Sudan (former)                     0.00
Ethiopia PDR                       0.00
Comoros                           59.84
Seychelles                       442.34
Sao Tome and Principe          12662.63
Cabo Verde                     14650.74
Guinea-Bissau                  19102.77
Lesotho                        21267.96
Botswana                       22101.30
Djibouti                       22729.91
Gambia                         23154.18
Gabon                          27979.64
Liberia                        29342.20
Namibia                        29874.89
Central African Republic       29937.00
Congo                          41181.68
Togo                           49841.88
Mauritius                      51114.83
Eswatini                       54343.33
Sierra Leone                   55311.33
Chad                           71594.68
Rwanda                         73663.69
Zimbabwe                       75919.34
Senegal                        95681.15
Guinea                         98138.87
Burkina Faso                  101855.07
Zambia                        103223.77
Tunisia                       124167.20
Benin                         124771.22
Niger                         126707.58
Madagascar                    131197.73
Mali                          149928.33
Mauritania                    156665.46
Mozambique                    161407.98
Malawi                        181098.71
Uganda                        213950.38
Côte d'Ivoire                 224599.01
Angola                        229159.57
Cameroon                      232030.43
Sudan                         239931.92
Kenya                         264660.66
United Republic of Tanzania   322616.85
Algeria                       325644.27
Ghana                         337599.06
Morocco                       388495.36
Ethiopia                      448683.76
South Africa                  517590.54
Egypt                         866379.92
Nigeria                      1483268.23

Answer: **Guinea-Bissau**